tests/src/JIT/HardwareIntrinsics/General/Shared/VectorGetAndWithElementTest.template

```diff
@@ -0,0 +1,134 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/******************************************************************************
+ * This file is auto-generated from a template file by the GenerateTests.csx  *
+ * script in tests\src\JIT\HardwareIntrinsics\General\Shared. In order to make    *
+ * changes, please update the corresponding template and run according to the *
+ * directions listed in the file.                                             *
+ ******************************************************************************/
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+
+namespace JIT.HardwareIntrinsics.General
+{
+    public static partial class Program
+    {
+        private static void {Method}{BaseType}{Imm}()
+        {
+            var test = new VectorGetAndWithElement__{Method}{BaseType}{Imm}();
+
+            // Validates basic functionality works
+            test.RunBasicScenario();
```

Comment:
We need senarios to catch `ArgumentOutOfRangeException` from `WithElement` and `GetElement`.

Reply:
Will update.

Reply:
Updated the template to have a `RunArgumentOutOfRangeScenario` that validates that both the regular and reflection code path throws for `{Imm} - ElementCount` and `{Imm} + ElementCount` (the former should be negative and the latter should be past the end of the vector).